### DISCUSSION TITLE
Can I pool parameter estimates of multiple imputed data obtained through10-fold cross-validation？

### DISCUSSION THREAD

**Discussion by shenlan17** [amices/mice#398] [Impute, analyse and pool]
I performed 10-fold cross-validation on mice，using `'ignore'` parameter in `mice` package, 5 datasets were obtained each time, and a total of 50 imputed datasets were obtained. I want to do linear regression using these 50 datasets  . 

Can I use Rubin's Rules to directly combine the effect estimates(β) of 50 datasets and then derive confidence intervals and p-values?  or only every 5 datasets ( imputed by the same training set and test set ) can be combined?

Thank you,
Liu.

**Comment by gerkovink** [comment]
@paulinavonstackelberg Could you shine your light on this, perhaps with a reference to code?

**Comment by paulinavonstackelberg** [comment]
Hi Liu,
could you elaborate a bit more on your goal? E.g., what are you interested in, and what is your evaluation measure?

**Reply by gerkovink** [reply]
This is a good remark. Depending on the aim, there could be different approaches to pooling the predictions, parameter estimates, etc.

**Comment by shenlan17** [comment]
Sorry for the confusion in the message.
 
Actually, I split my whole dataset into two parts: the training dataset (80% observations) and the testing dataset (the other 20% observations). Now, I would like to use `mice` package to impute the missing value in the training dataset, and then use the same model to impute the missing values in the test dataset. The `ignore` option in mice function was used, and 5 sets of data were imputed.

```
#The dataset has 100 observations and 20 variables #
ig<-c(rep(FALSE,80),rep(TRUE,20))
imp <- mice(data, m=5, maxit=10, seed=1234, ignore=ig)
imp.data=complete(imp,'all')
```

Subsequently, I perform multiple linear regression in the imputed training dataset and testing dataset (i.e, with the same variables but different observations), respectively. 
 
```
for (i in 1:5){                         #5 imputed data
     training.data=imp.data[[ i ]][1:80,]
     training.data=imp.data[[ i ]][81:100,]
     lm(y~.，data=training.data)
}
```

Here, I am just wondering that could I direct use the Rubin's Rules to combine the effect estimates (i.e., beta coefficients) and derive the p values in the 5 training datasets, 5 testing datasets or 10 datasets (including all training and testing datasets)? In other words, if the observations of two datasets (here, training and testing) are different, could I use Rubin's Rules to combine the effect estimates?
 
 
Second question, following the first question, I also want to split the whole observations 10 times, i.e., random sample 80% observations as training dataset and the other 20% as testing dataset in each time. After imputation, I totally obtain 50 training dataset and 50 testing datasets. I combine each training dataset with the corresponding testing dataset, and then get 50 imputed whole datasets. Subsequently, linear regression is performed in each of these 50 imputed whole datasets. I am wondering could I use Rubin's Rules to combine the effect estimates in these 50 imputed whole datasets.

```
set.seed(1234)
folds<-createFolds(1:dim(data)[1],k=10)
for(ff in 1:length(folds)){
         testindex <- folds[[ff]]
         ig<-rep(FALSE,dim(data)[1])
         ig[testindex]<-TRUE 
         imp <- mice(data1,m=5,seed = 1,maxit=5,ignore = ig)
  }
```

Thank you
